server/routes/induction/common/wantToAddQualificationsController.ts

```diff
@@ -27,8 +27,9 @@ export default abstract class WantToAddQualificationsController extends Inductio
 
     const functionalSkills = {
       ...prisonerFunctionalSkills,
-      assessments: mostRecentAssessments(prisonerFunctionalSkills.assessments || []),
+      assessments: mostRecentAssessments(prisonerFunctionalSkills?.assessments || []),
```

Comment:
Hmmm, I _think_ you should always have a `prisonerFunctionalSkills` object (its data that comes from Curious). I _think_ the object itself should always exist, but the `assessments` array within it might not ... 🤔 

Given that you've had to make this change, I wonder if we've forgotten something in the route setup somewhere ... let me check ....

Reply:
Yes, I was right; instead of this change we need to add a call to a middleware, exactly as we do for the page on the update journey:
```
    router.get('/prisoners/:prisonNumber/induction/want-to-add-qualifications', [
      retrieveFunctionalSkillsIfNotInSession(services.curiousService),
      asyncMiddleware(wantToAddQualificationsUpdateController.getWantToAddQualificationsView),
    ])
```

I think if you add that middleware call to the route you've created, you can undo the changes to this file 👍